transport/server.cc

```diff
@@ -997,19 +997,16 @@ cql_server::connection::process(uint16_t stream, request_reader in, service::cli
         tracing::trace_state_ptr trace_state, Process process_fn) {
     fragmented_temporary_buffer::istream is = in.get_stream();
 
-    return process_fn(client_state, _server._query_processor, in, stream,
-            _version, permit, trace_state, true, {})
-            .then([stream, &client_state, this, is, permit, process_fn, trace_state]
-                   (cql_server::process_fn_return_type msg) mutable {
+    auto msg = co_await process_fn(client_state, _server._query_processor, in, stream,
+            _version, permit, trace_state, true, {});
+        // FIXME: indentation
```

Comment:
How is this related to dropping permit?

Reply:
Since we don't pass the permit down to process_on_shard we need to hold on to the permit passed to us.
Making this function a coroutine achieves that.
I can do the coroutinization in a preliminary patch if needed.

Reply:
At least explain it in the changelog

Reply:
done in v4